kubemarine/system.py

```diff
@@ -152,33 +96,6 @@ def enrich_kernel_modules(inventory: dict, cluster: KubernetesCluster):
 
     return inventory
 
-def get_system_packages_for_upgrade(cluster):
-    upgrade_ver = cluster.context["upgrade_version"]
-    previous_ver = cluster.context["initial_kubernetes_version"]
-    compatibility = cluster.globals["compatibility_map"]["software"]
-
-    # handle special cases in which upgrade is not required for particular package
-    cluster_associations = cluster.inventory["services"]["packages"]["associations"][cluster.get_os_family()]
-    upgrade_associations = cluster.procedure_inventory.get(upgrade_ver, {}).get("packages", {}).get("associations", {})
-    system_packages = get_system_packages(cluster)
-    upgrade_required = list(system_packages)
-    for package in system_packages:
-        defined_association = upgrade_associations.get(package, {}).get("package_name")
-        if defined_association and defined_association == cluster_associations[package]['package_name']:
-            # case 1: package_name is defined in upgrade inventory but is equal to one already defined in cluster.yaml
-            upgrade_required.remove(package)
-        elif compatibility.get(package) and compatibility[package][upgrade_ver] == compatibility[package][previous_ver] \
```

Comment:
This had some problems, for example if `containerdio` has different version for old and new Kubernetes versions, the CRI was not upgraded for clusters on RHEL and containerd.

I fixed this, but this leads to CRI upgrade if `podman` is upgraded. Though if fact it is not necessary to perform the regular upgrade procedure as podman does not manage Kubernetes containers.

I believe that such new redundancy is acceptable. I see possible fix of the problem to exclude podman from packages.associations for containerd.